### ISSUE TITLE
Shishkina Isobars not got right curvature vs. ones in their paper.

### ISSUE THREAD

**Issue by PennyWieser** [kaylai/VESIcal#100]
Could have sworn I posted this yesterday, though apparently not!
Shishkina is looking really good, except the isobars don't match those in their figure in the paper - the colored lines should match the black lines. 
![image](https://user-images.githubusercontent.com/65668673/87064141-35b78280-c207-11ea-9fc7-488cea957135.png)

See Testing_Shishkina_et_al_2014.pynb and xlsx.

**Comment by simonwmatthews** [comment]
The thing that looks most worrying to me here is the model seems to be predicting the wrong CO2 concentrations at the low-H2O terminus of the curves? Am I understanding that correctly? If that's the case, it's probably a bug which I should be able to find.

The curvature of the line depends on how the model handles mixing. For this model I've assumed ideal mixing in VESIcal (in both the gas and magma) because, as far as I can tell, the original manuscript doesn't say anything about the assumptions they make for the mixed fluid calculations. In fact, I ended up assuming their mixed fluid curves were sketched in, rather than calculated... This is *probably* incorrect though. In other models (e.g. Dixon), the different curvature comes from the fugacity model- where the non-ideality of both the gases and the mixing is accounted for. Shishkina doesn't use a fugacity model though...

Perhaps it's worth contacting them?

**Comment by simonwmatthews** [comment]
Oh, and are the curves starting from the wrong maximum H2O content too?

**Comment by PennyWieser** [comment]
Okay, part the problem here was I copied the data from a table in the supplement with FeO, Fe2O3, and FeOT headings where only FeOT had anything in. VESIcal treats this as if there is no FeO at all, hence some of the compositions were well off. But having sorted that, they are still off.
![image](https://user-images.githubusercontent.com/65668673/87141420-ec187780-c29a-11ea-9b14-cfeefdd03ad0.png)

If I use the pi star and Pi values in their table for the compositions they say they are plotting on this graph , and calculate CO2 conc using their equation lnCO2=Alnp.... etc., I get pretty much the same answers as VESIcal - So I have no idea how they calculated these isobars, as nothing I can do can get them to match for pure CO2 fluids.

**Comment by kaylai** [comment]
> Okay, part the problem here was I copied the data from a table in the supplement with FeO, Fe2O3, and FeOT headings where only FeOT had anything in. VESIcal treats this as if there is no FeO at all, hence some of the compositions were well off. 

Hmm I thought I fixed this for batch calculation methods at least. I think it's because I had accounted for FeOt, FeOtot, FeOtotal, FeOstar, and FeO*, but not FeOT. I've added in FeOT to the list of "FeOtotal" type columns to catch. If one of these is passed AND FeO is not passed, a warning is shown to let the user know that that column will be used as FeO (FeO/Fe2O3 not calculated). If both FeO and some FeOtotal type column is passed, a warning is written to let the user know that the FeO column will be used, and the FeOtotal type column will be ignored.

Commit 1dadaa608560d262ebe4825c7a9aef54b04a0d5b
